src/Modal/Modal.tsx

```diff
@@ -0,0 +1,236 @@
+     * This is useful when the dialogue would be inside a scrollable container or one with "overflow: hidden"
+     * so it doesn't get cut off. Uses IntersectionObserver and needs a polyfill if IE compatibility is needed. This
+     * defaults to `false`.
+     */
+    renderInPortal?: boolean;
+    /**
+     * The classes to pass to the modal.
+     */
+    className?: string;
+}
+
+/**
+ * Modal Component
+ *
+ * The Modal component is a UI blocking dialogue overlay.
+ * The state is not managed inside this component and visibility (via the `isOpen` prop) needs to be maintained in the parent component.
+ * While the default rendering behavior is often sufficient, the `renderInPortal` prop can be used
+ * to append a div to the body.
+ *
+ */
+const Modal: FC<ModalProps> = (props) => {
+    const {
+        handleClose,
+        isOpen,
+        headerText = '',
+        isCloseIconVisible = true,
+        actionButton,
+        cancelButtonText = CANCEL_BUTTON_TEXT,
+        subHeaderText = '',
+        children,
+        showTopDivider = false,
+        showBottomDivider = false,
+        dialogWidth = 'sm',
+        renderInPortal = false,
+        className
+    } = props;
+
+    const handleOnClose = () => {
+        if (handleClose) {
+            handleClose();
+        }
+    };
+
+    const [portal, setPortal] = useState<HTMLElement | null>(null);
+    const [dialogElement, setDialogElement] = useState<HTMLElement | null>(null);
+    const [update, setUpdate] = useState<number>(0);
+
+    useEffect(() => {
+        const bodyElementHTMLCollection = document.getElementsByTagName('body');
+        const bodyElement = bodyElementHTMLCollection.length > 0 ? bodyElementHTMLCollection.item(0) : null;
+        let observer: IntersectionObserver;
+        let portalElement: HTMLElement;
+
+        if (renderInPortal && bodyElement) {
+            portalElement = document.createElement('div');
+            portalElement.onclick = (e) => {
+                if (e.target !== portalElement) {
+                    return;
+                }
+                handleOnClose();
+            };
+
+            if (!portal) {
+                bodyElement.appendChild(portalElement);
+            }
+
+            if (!portal && dialogElement) {
+                observer = new IntersectionObserver(() => {
+                    setUpdate(new Date().getTime());
+                });
+
+                observer.observe(dialogElement);
+                setPortal(portalElement);
+            }
+        }
+
+        return () => {
+            if (dialogElement && observer) {
+                observer.unobserve(dialogElement);
+            }
+
+            if (portalElement && bodyElement && bodyElement.contains(portalElement)) {
+                bodyElement.removeChild(portalElement);
+            }
+        };
+    }, [dialogElement]);
+
+    useEffect(() => {
+        if (dialogElement && portal) {
```

Comment:
We can probably remove this whole useEffect, the update state, as well as the IntersectionObserver. I don't think it will ever fire, and if it does these styles might make the modal look wrong.

Reply:
Actually it breaks when you try to remove any of them --- those styles need to get set on the portal in order for it to work properly. With that said, I noticed I did have the wrong `justifyContent` set, but the rest of the styles mimic the `DialogContainer` styling.

Reply:
Sounds good.